gearoenix/math/gx-math-aabb.hpp

```diff
@@ -124,39 +124,67 @@ struct Aabb3 final {
     [[nodiscard]] constexpr std::optional<Element> hit(const Ray3<Element>& ray, const Element d_min) const noexcept
     {
         const auto& ro = ray.get_origin();
+        const auto& rd = ray.get_normalized_direction();
         const auto& rrd = ray.get_reversed_normalized_direction();
         
-        if (fabs(rrd.x) < FLT_EPSILON)
+        const Element tmin = (Element)0.0;
+        const Element tmax = (Element)FLT_MAX;
+
+        if (fabs(rd.x) < FLT_EPSILON)
```

Comment:
We have an epsilon in the `gx-cr-types.hpp`.

Reply:
Usually, choice of epsilon value should depend on the range of input data and one specific value can't be picked for the entire range of floating-point values.

Here we're dealing with normalized ray directions which means floating-point values in the [0-1] range. 1e-4 (GX_POS_EPSILON) might be too big here.

Reply:
Yeah, exactly this is the good thing about the `GX_POS_EPSILON`, the user later can change its value based on the project it's being used.

Reply:
Or we can have an override that gets epsilon in its arguments.

Reply:
Even across one project, one single Epsilon shouldn't be used for every situation. User should pick a "reasonable" value based on the specific computation being done and the range of input values for that computation. A project-wide defined epsilon isn't the way to go IMO.  For example. 1e-4 might be too big for the [0-1] range, but it also might be too small for values in the [32768-65536] range (precision is about 0.0004 in that range)